### PR TITLE
Set Up Continuous Integration Through Github Actions

### LINKED ISSUE #107: Set Up CI and CD
CI will be done through Github Actions, ensuring that linting and tests will be checked upon each merge into master.
CD will be done through Github Pages, to be integrated with Github actions for continuous deployment.

### PR DESCRIPTION
**Pull request by lucas2005gao**
CI implemented here through Github Actions, ensuring that CD through Github Pages will be updated upon each PR.
Linter checks and Tests will be done in other Github Action workflows

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import React from "react";
-import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
+import { HashRouter as Router, Switch, Route } from "react-router-dom";
```
Comment (lucas2005gao) [inline src/App.js:2]: we need to change to HashRouter here because GitHub pages doesn't work with BrowserRoute

**Review by max-gurr — APPROVED** [review]
Looks good! Thanks for setting this up :)

**Comment by lucas2005gao** [comment]
> Looks good! Thanks for setting this up :)

Oops sorry, I accidently re-requested your review cuz it was top on suggested reviewers lmao

**Review by max-gurr — COMMENTED** [review]
Hehe no worries, here's another one

**Pull request merged**